packages/nitro-protocol/contracts/ninja-nitro/readme.md

```diff
@@ -0,0 +1,97 @@
+
+## Overview
+
+### Recap of Mono-Nitro
+
+The main implementation has 1 x monolithic `Adjudicator` contract and m x `AssetHolder` contracts (one for each of the m Assets in the state channel network). The `Adjudicator` API is
+
+- The ForceMove API
+- `pushOutcome`
+- utility methods that just batch up this API and the `AssetHolder` API (`pushOutcomeAndTransferAll` and `concludePushOutcomeAndTransferAll`)
+
+The essence of the architecture, then, is `pushOutcome`: a finalized outcome must be "pushed" from the `Adjudicator`, resulting in a piece of that outcome being stored in each `AssetHolder`. Nevertheless, `pushOutcome` is an implementation detail and not necessary part of the overall Nitro API.
+
+The `AssetHolder` API is:
+
+- `transfer`
+- `claim`
+- permissioned methods `transferAllAdjudicatorOnly` and `setAssetOutcomeHash` only callable from the `Adjudicator`.
+
+After a `pushOutcome` ssets may be released by targetting each `AssetHolder` and calling `claim` or `transfer` (which _are_ essential parts of the overall Nitro API). This is the unhappy path. The happy path will shortcut `pushOutcome` and use the permissioned methods.
+
+### Ninja-Nitro
+
+Ninja-Nitro, by contrast has 1 x `AdjudicatorFactory` contract, 1 x `SingleChannelAdjudicator` master contract, and k x `Proxy` contracts, one for each of the k channels in the network that resolves on chain. The factory and master contracts are "infastructure" and deployed once (most likely by the protocol team).
+
+The `AdjudicatorFactory` API is
+
+- `createChannel`
+- utility methods that batch `createChannel` with the master contract API.
+
+The `createChannel` method will deploy a new `Proxy` for each channel as necessary (namely, if it needs to be directly defunded). Such `Proxy` contracts are lightweight (cheap to deploy), have easily predictable addresses, and point to the `SingleChannelAdjudicator` master contract. They therefore have the same API:
+
+- The ForceMove API
+- `transfer`
+- `claim`
+- utility methods that batch up this API
+
+Note however, that the Proxy contracts each have their own storage. See https://eips.ethereum.org/EIPS/eip-1167.
+
+## Similarities and differences
+
+### Holdings
+
+The Mono-Nitro system has m x `holdings` mappings, one in each AssetHolder. This mapping is `public`, so has a public getter function.
+Ninja-Nitro has no such mappings: but a public `holdings` function remains. This functions simply reads the balance of the current contract (or another contract) from either i) the Ethereum accounting system or ii) a given Token contract.
+
+### Deposits
+
+In Mono-Nitro, depositing requires calling a method on each `AssetHolder`.
+In Ninja-Nitro, depositing can be achieved simply by sending funds or tokens to the address of the `Proxy`, and this can even be done before the `Proxy` is created/deployed.
```

Comment:
Can this be done after the `Proxy` is destroyed?

Reply:
Yes I believe so. I can add something to our test suite to show that, and an extra comment here.

Reply:
75d5c94